src/file_organizer/undo/rollback.py

```
@@ -276,6 +284,76 @@ def rollback_copy(self, operation: Operation) -> bool:
+                link_stat.st_ino,
+            ):
+                logger.error(
+                    "Refusing to rollback hardlink operation; destination no longer "
+                    f"matches source inode: {link_path}"
+                )
+                return False
+            link_path.unlink()
+            fsync_directory(link_path)
+            logger.info(f"Successfully rolled back hardlink operation {operation.id}")
+            return True
+        except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)